pubsub/gcloud/aio/pubsub/subscriber_client.py

```diff
@@ -100,18 +118,30 @@ def _callback_wrapper(message: Message) -> None:
             return _callback_wrapper
 
 else:
```

Comment:
Now that we're turning this into a proper http interface rather than the grpc pass-through thing, you should be able to do away with a lot of these conditionals and build this for both `gcloud-aio` and `gcloud-rest` with the same code!

Reply:
Do you think we can get rid of having `pass-through` thing for `gcloud-rest`? That will lock anyone using it to the current version. I'm all for it, just double checking :)

Reply:
Yeah, I think we could go ahead and release this whole thing as a major version bump -- totally reasonable for folks to have to pin to the current major until they're ready to change the approach to match :)